lib/apiservers/engine/backends/convert/stats.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package convert
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/docker/docker/api/types"
+
+	"github.com/vmware/vic/lib/portlayer/metrics"
+)
+
+// ContainerStats encapsulates the conversion of VMMetrics to
+// docker specific metrics
+type ContainerStats struct {
+	config         ContainerStatsConfig
+	dockerStats    *types.StatsJSON
+	currentMetrics *metrics.VMMetrics
+	totalVCHMhz    uint64
+	dblVCHMhz      uint64
+	preTotalMhz    uint64
+
+	// reader/writer for stream
+	reader *io.PipeReader
+	writer *io.PipeWriter
+}
+
+type ContainerStatsConfig struct {
+	Ctx         context.Context
+	Cancel      context.CancelFunc
+	Out         io.Writer
+	ContainerID string
+	Stream      bool
+	VchMhz      int64
+}
+
+type InvalidOrder struct {
```

Comment:
We should rename `InvalidOrder` to `InvalidOrderError` since it's an error type.